rfcs/QueryLevelNullability.md

```diff
@@ -0,0 +1,179 @@
+
+## 🧑‍💻 Proposed syntax
+
+It would make more sense if the client could express that `name` must be non-null _in the query itself_:
+```graphql
+query GetBusinessName($encid: String!) {
+  business(encid: $encid) {
+    name!  <-- this!
+  }
+}
+```
+Semantically the GraphQL `!` operator is nearly identical to its counterpart in Swift (also represented by `!`) which is
+referred to as the "force unwrap operator". In Swift, for example, you can cast a string to an integer with `Int("5")` 
+but the string being cast may not be a valid number, so that statement will return `null` rather than an integer if the
+string cannot be turned into an integer. If you want to ensure that the statement does not return `null` you can instead 
+write `Int("5")!`. If you do that, an exception will be thrown if the statement would return `null`.
+
+In GraphQL, the `!` operator will act similarly. In the case that `name` does not exist, the query will return an
+error to the client rather than any data.
+
+On web where codegen is not used, the client no longer needs to handle the case where expected fields are missing.
+On mobile platforms where codegen is used, clients have full control over the nullability of the properties on the
+generated types. Since nullability is expressed in the query rather than the schema, it's flexible enough to accommodate
+various use-cases (e.g., where the business `name` _is_ allowed to be nullable).
+
+### `!`
+
+We have chosen `!` because `!` is already being used in the GraphQL spec to indicate that a field is non-nullable.
+Incidentally the same precedent exists in Swift (`!`) and Kotlin (`!!`) which both use similar syntax to indicate
+"throw an exception if this value is `null`". 
+
+### Use cases
+
+#### When a field is necessary to the function of the client
+There are times where a field can be `null`, but a feature that queries for the field will not function
+if it is `null`. For example if you are trying to render an information page for a movie, you won't
+be able to do that if the name field of the movie is missing. In that case it would be preferable
+to fail as early as possible.
+
+### ✨ Examples
+
+#### Non-nullable field
+```graphql
+query GetBusinessName($encid: String!) {
+  business(encid: $encid) {
+    name!
+  }
+}
+```
+would codegen to the following type on iOS.
+
+```swift
+struct GetBusinessNameQuery.Data.Business {
+  let name: String // lack of `?` indicates that `name` will never be `null`
+}
+```
+
+#### Non-nullable object with nullable fields
+```graphql
+query {
+  business(id: 4) {
+    reviews! {
+      rating
+      text
+    }
+  }
+}
+```
+
+## ✅ RFC Goals
+- Non-nullable syntax that is based off of syntax that developers will already be familiar with
+
+## 🚫 RFC Non-goals
+
+## 🗳️ Alternatives considered
+
+### Make Schema Fields Non-Nullable Instead
+Discussion on [this topic can be found here](https://medium.com/@calebmer/when-to-use-graphql-non-null-fields-4059337f6fc8)
```

Comment:
TODO: Flesh this out with things about versioning, breaking changes, and the impact it has on partial results